src/main/java/org/mitre/synthea/export/rif/BeneficiaryExporter.java

```diff
@@ -0,0 +1,471 @@
+package org.mitre.synthea.export.rif;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.mitre.synthea.helpers.Config;
+import org.mitre.synthea.helpers.RandomCollection;
+import org.mitre.synthea.helpers.Utilities;
+import org.mitre.synthea.world.agents.Person;
+
+/**
+ * Export beneficiary and beneficiary history files.
+ */
+public class BeneficiaryExporter extends RIFExporter {
+
+  static String getBB2SexCode(String sex) {
+    switch (sex) {
+      case "M":
+        return "1";
+      case "F":
+        return "2";
+      default:
+        return "0";
+    }
+  }
+
+  /**
+   * Construct an exporter for Beneficiary and BeneficiaryHostory file.
+   * @param startTime earliest claim date to export
+   * @param stopTime end time of simulation
+   * @param exporter the exporter instance that will be used to access code mappers
+   */
+  public BeneficiaryExporter(long startTime, long stopTime, BB2RIFExporter exporter) {
+    super(startTime, stopTime, exporter);
+  }
+
+  /**
+   * Export a beneficiary details for single person.
+   * @param person the person to export
+   * @return the beneficiary ID
+   * @throws IOException if something goes wrong
+   */
+  public String export(Person person) throws IOException {
+    String beneIdStr = Long.toString(RIFExporter.nextBeneId.getAndDecrement());
+    person.attributes.put(RIFExporter.BB2_BENE_ID, beneIdStr);
+    String hicId = RIFExporter.nextHicn.getAndUpdate((v) -> v.next()).toString();
+    person.attributes.put(RIFExporter.BB2_HIC_ID, hicId);
+    String mbiStr = RIFExporter.nextMbi.getAndUpdate((v) -> v.next()).toString();
```

Comment:
It is kind of unfortunate that the HICN and MBI need to pass a "lambda" to the `getAndUpdate` method, rather than do something like `RIFExporter.nextBeneId.getAndDecrement()`... but I suppose it can't be avoided unless you want to just wrap that in a method call.